src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj

```diff
@@ -74,11 +75,35 @@
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.SharedFramework.Sdk" />
 
   <Target Name="ResolveReadyToRunCompilers" DependsOnTargets="ResolveRuntimeFilesFromLocalBuild">
+    <PropertyGroup> <!-- Can be removed once runtime repo switches over to 6.0 drop of the SDK repo -->
+      <JitTargetArchComponent>$(TargetArchitecture)</JitTargetArchComponent>
+      <JitTargetOSComponent Condition="'$(TargetOS)' == 'windows'">win</JitTargetOSComponent>
+      <JitTargetOSComponent Condition="'$(TargetOS)' == 'linux'">unix</JitTargetOSComponent>
+      <JitTargetOSComponent Condition="'$(TargetOS)' == 'osx'">osx</JitTargetOSComponent>
```

Comment:
I think we never use `osx`, do we? See crossgen2.csproj: https://github.com/dotnet/runtime/blob/69e114c1abf91241a0eeecf1ecceab4711b8aa62/src/coreclr/tools/aot/crossgen2/crossgen2.csproj#L64-L65

Reply:
Hmm, you're right for sure, we should produce identical code when targeting Linux and OSX. In light of this I'm wondering why I'm seeing "clrjit_unit_osx_arm64_x64.dll" in my crossgen2 folder - @davidwrighton, can you please explain to me what that stands for?

Reply:
At present we use `unix_osx` for OSX on ARM64. You can ignore that for now. #45467

Reply:
For your primary comment, I have addressed that in 2nd commit along with a few additional bugs on Linux as I traditionally debugged this on Windows first ;-).